container-engine-lib/lib/backend_impls/remote_context_backend/remote_context_kurtosis_backend.go

```diff
@@ -0,0 +1,208 @@
+package remote_context_backend
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/api_container"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/enclave"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/engine"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/exec_result"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_collector"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_database"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/networking_sidecar"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/stacktrace"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"net"
+)
+
+// RemoteContextKurtosisBackend is a dual context holding a reference to a local backend running on Docker (k8s is
+// currently not supported to work with a remote Kurtosis backend) and a reference to a remote Kurtosis backend (which
+// currently is a Docker backend, we do not support anything else).
+// Basically, requests are dispatched among those 2 backends depending on which area of Kurtosis they touch.
+// This might change, but right now the engine runs locally, therefore engine interacting endpoint goes to the local
+// backend, and all the rest goes to the remote backend, as all other containers (including API container) should be
+// running in the remote backend
+type RemoteContextKurtosisBackend struct {
```

Comment:
How does this work with MetricsReportingKurtosisBackend? Tbh we are now only reporting metrics from the CLI and if this doesn't support it I am happy with a banner saying it doesn't support; I don't want to you to code the support just curious

Reply:
I don't think this is incompatible with MetricsReportingKurtosisBackend right now. It works the exact same as a local docker backend. Do you think it's wrong? Whatever the backend is, I was thinking we could track the metric in a similar way

Reply:
And yes, definitely if we want to move the engine remotely, then no need for local docker anymore. That being said, I think we're more tracking keeping some things locally and making that work with the others things running remotely. But that could change. IDK if requiring Docker is a big blocker to adoption